payjoin-cli/src/app/wallet.rs

```diff
@@ -23,6 +23,9 @@ pub struct BitcoindWallet {
 impl BitcoindWallet {
     pub fn new(config: &crate::app::config::BitcoindConfig) -> Result<Self> {
         let client = match &config.cookie {
+            Some(cookie) if cookie.as_os_str().is_empty() => {
+                return Err(anyhow!("Cookie authentication enabled but no cookie path provided in config.toml"))
+            }
```

Comment:
At first I considered that this validation ought to happen in the config.rs module, however, the cookie details and wallet impl here are tightly coupled and it would be even more code in the config mod.